plugins/linkattacher/src/linkattacherinterface.cpp

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (C) 2013-2015 Fondazione Istituto Italiano di Tecnologia RBCS & iCub Facility & ADVR
+ * Authors: see AUTHORS file.
+ * CopyPolicy: Released under the terms of the LGPLv2.1 or any later version, see LGPL.TXT or LGPL3.TXT
+ */
+
+#include <linkattacherinterface.h>
+
+using namespace gazebo;
+using namespace std;
+using namespace yarp::os;
+
+LinkAttacherInterface::LinkAttacherInterface() : ModelPlugin(), m_network(0), m_rpcport(0)
+{
+}
+
+LinkAttacherInterface::~LinkAttacherInterface()
+{
+  if(m_rpcport)
+  {
+    m_rpcport->close();
+    delete m_rpcport;
+  }
+
+  if(m_network)
+    delete m_network;
+}
+
+void LinkAttacherInterface::Load(physics::ModelPtr _model, sdf::ElementPtr _sdf)
+{
+  if(m_network!=0)
+    return;
+
+  m_network = new yarp::os::Network();
+
+  if(!yarp::os::Network::checkNetwork(GazeboYarpPlugins::yarpNetworkInitializationTimeout))
+  {
+    yError() << LogPrefix << "Load: yarp network does not seem to be available, is the yarpserver running?";
+    return;
+  }
+
+  std::string model_name = _model->GetName();
+
+  yInfo() << LogPrefix << "Load: Model name: " << model_name;
+
+  m_la_server.attachWorldPointer(_model->GetWorld());
+  m_la_server.attachModelPointer(_model);
+
+  bool configuration_loaded = false;
+
+  if( _sdf->HasElement("yarpConfigurationFile") )
```

Comment:
I'm not sure, but shouldn't you look for `plugin name='link_attacher'` and then inside of it `yarpConfigurationFile`? Or the scope of `_sdf` is already the plugin scope?